src/OpenTelemetry/Metrics/Export/PushMetricController.cs

```diff
@@ -0,0 +1,120 @@
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Metrics.Configuration;
+using static OpenTelemetry.Metrics.Configuration.MeterFactory;
+
+namespace OpenTelemetry.Metrics.Export
+{
+    internal class PushMetricController
+    {
+        private readonly TimeSpan pushInterval;
+        private readonly Task worker;
+        private MetricExporter metricExporter;
+        private MetricProcessor metricProcessor;
+        private Dictionary<MeterRegistryKey, MeterSdk> meters;
+
+        public PushMetricController(Dictionary<MeterRegistryKey, MeterSdk> meters,
+            MetricProcessor metricProcessor,
+            MetricExporter metricExporter,
+            TimeSpan pushInterval,
+            CancellationTokenSource cts)
+        {
+            this.meters = meters;
+            this.metricProcessor = metricProcessor;
+            this.metricExporter = metricExporter;
+            this.pushInterval = pushInterval;
+            this.worker = Task.Factory.StartNew(
+                s => this.Worker((CancellationToken)s), cts.Token);
+        }
+
+        private async Task Worker(CancellationToken cancellationToken)
+        {           
+            List<Metric<long>> longMetricToExport = new List<Metric<long>>();
+            List<Metric<double>> doubleMetricToExport = new List<Metric<double>>();
+
+            await Task.Delay(this.pushInterval, cancellationToken).ConfigureAwait(false);
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                var sw = Stopwatch.StartNew();
+                try
+                {                    
+                    longMetricToExport.Clear();
+                    doubleMetricToExport.Clear();
+
+                    foreach (var meter in this.meters.Values)
+                    {
+                        meter.Collect();
+                    }
+
+                    OpenTelemetrySdkEventSource.Log.CollectionCompleted(sw.ElapsedMilliseconds);
+
+                    // Collection is over at this point. All metrics are given
+                    // to the MetricProcesor(Batcher).
+                    // Let MetricProcessor know that this cycle is ending,
+                    // and send the metrics from MetricProcessor
+                    // to the MetricExporter.
+                    var metricsToExportTuple = this.metricProcessor.FinishCollectionCycle();
+                    if (metricsToExportTuple.Item1 != null)
+                    {
+                        longMetricToExport.AddRange(metricsToExportTuple.Item1);
+                    }
+
+                    if (metricsToExportTuple.Item2 != null)
+                    {
+                        doubleMetricToExport.AddRange(metricsToExportTuple.Item2);
+                    }
+
+                    var longExportResult = await this.metricExporter.ExportAsync<long>(longMetricToExport, cancellationToken);
```

Comment:
nit: maybe here we can check the `cancellationToken` before the call to exporter? Just in case everything was cancelled while we were collecting metrics? This method may be slow.